erasure-readfile.go

```diff
@@ -97,25 +99,34 @@ func erasureReadFile(disks []StorageAPI, volume string, path string, partName st
 
 		// Check blocks if they are all zero in length, we have corruption return error.
 		if checkBlockSize(enBlocks) == 0 {
-			return nil, errXLDataCorrupt
+			return bytesWritten, errXLDataCorrupt
 		}
 
 		// Verify if reconstruction is needed, proceed with reconstruction.
 		if !noReconstruct {
 			err := decodeData(enBlocks, eInfo.DataBlocks, eInfo.ParityBlocks)
 			if err != nil {
-				return nil, err
+				return bytesWritten, err
 			}
 		}
 
 		// Get data blocks from encoded blocks.
 		dataBlocks, err := getDataBlocks(enBlocks, eInfo.DataBlocks, int(curBlockSize))
 		if err != nil {
-			return nil, err
+			return bytesWritten, err
 		}
 
 		// Copy data blocks.
-		buffer = append(buffer, dataBlocks...)
+		var n int64
+		if offset > 0 {
+			n, err = io.Copy(writer, bytes.NewBuffer(dataBlocks[offset:]))
+		} else {
+			n, err = io.Copy(writer, bytes.NewBuffer(dataBlocks))
+		}
+		bytesWritten += int64(n)
+		if err != nil {
+			return bytesWritten, err
```

Comment:
Another problem is you are not taking in the 'length' requirement you are writing more than requested.